packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -393,6 +398,15 @@ export class IndexedDbPersistence implements Persistence {
               oldestChangeId
             );
           }
+        })
+        .next(() => {
+          // Delete potential leftover entries that may continue to mark the
+          // inactive clients as zombied in LocalStorage.
+          inactiveClients.forEach(inactiveClient => {
+            this.window.localStorage.removeItem(
+              this.zombiedClientLocalStorageKey(inactiveClient.cliendId)
+            );
+          });
```

Comment:
FWIW, this would be slightly more leakproof if you did this before deleting the IndexedDb entries.  (theoretically a crash after deleting the IndexedDb entries but before deleting the LocalStorage entry would leave the LocalStorage entries indefinitely).  I don't think I really care though.

Reply:
Yes, that's true. But then the IndexedDb entries could come back to life, and I definitely don't want that.

Reply:
Hm, I guess I need to do that after committing the transaction. I'll ponder about that some more.